playground/models/networks/sampling.py

```diff
@@ -1,4 +1,20 @@
-from keras.layers import Dense, Concatenate, Add
+from keras.layers import Dense, Concatenate, Add, Lambda, Reshape
+from keras import backend as K
+from numpy import newaxis
+
+
+def outer_product(inputs):
+    """
+    inputs: list of two tensors (of equal dimensions,
+        for which you need to compute the outer product
+    """
+    x, y = inputs
+    feature_size = x._shape_as_list()[1] ** 2
+    outer_Product = x[:,:, newaxis] * y[:,newaxis,:]
```

Comment:
Don't want to be too pedantic but I find it better if a space is left after a `,` in a slicing operation.